test/libsolidity/semanticTests/array/copying/array_of_function_external_storage_to_storage_dynamic.sol

```diff
@@ -4,8 +4,8 @@ contract C {
     function testFunction3() public {}
 
 
-    function() external [3] externalArray0;
-    function() external [3] externalArray1 = [
+    function() external [] externalArray0 = new function() external[] (3);
+    function() external [] externalArray1 = [
         this.testFunction1,
         this.testFunction2,
         this.testFunction3
```

Comment:
I'm actually really surprised that this works. I suggested changing it to a static array in the original PR because I thought we never allow assigning array literals to dynamic arrays but turns out that we do allow that for storage. Probably because we're not assigning a reference but just copying the content.

Reply:
Would it be better if I add the assignment part into a constructor ?

Reply:
No, it's fine. I suggested that only because I thought it was not possible to do it here. But I wasn't sure - I assumed that if I'm wrong you'll notice that the CI passes and stay with that version or comment something :)

Reply:
Ohh, well, for precaution's sake, I already merged changes where I shifted the assignments to a Constructor. Hope that's fine :)

Reply:
It's fine. I liked the constructor-less version better but it's not like it's a big difference. It can stay as is.

What about https://github.com/ethereum/solidity/pull/12372#discussion_r769964255? It's a tiny change but that's how we format conditions in `return` and is an easy detail to change.

Reply:
Added the changes in the latest push